src/Features/CSharp/Portable/DocumentHighlighting/CSharpDocumentHighlightsService.cs

```diff
@@ -1,32 +1,31 @@
 // Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
 
-using System.Collections.Generic;
+using System.Collections.Immutable;
 using System.Composition;
 using System.Linq;
 using System.Threading;
 using System.Threading.Tasks;
 using Microsoft.CodeAnalysis.CSharp.Syntax;
-using Microsoft.CodeAnalysis.Editor.Implementation.ReferenceHighlighting;
+using Microsoft.CodeAnalysis.DocumentHighlighting;
 using Microsoft.CodeAnalysis.Host.Mef;
-using Roslyn.Utilities;
 
-namespace Microsoft.CodeAnalysis.Editor.CSharp.ReferenceHighlighting
+namespace Microsoft.CodeAnalysis.CSharp.DocumentHighlighting
 {
-    [ExportLanguageService(typeof(IReferenceHighlightingAdditionalReferenceProvider), LanguageNames.CSharp), Shared]
-    internal class ReferenceHighlightingAdditionalReferenceProvider : IReferenceHighlightingAdditionalReferenceProvider
+    [ExportLanguageService(typeof(IDocumentHighlightsService), LanguageNames.CSharp), Shared]
+    internal class CSharpDocumentHighlightsService : AbstractDocumentHighlightsService
     {
-        public async Task<IEnumerable<Location>> GetAdditionalReferencesAsync(
+        protected override async Task<ImmutableArray<Location>> GetAdditionalReferencesAsync(
```

Comment:
this code moved from the C#' "ReferenceHighlightingAdditionalReferenceProvider " directly into CSharpDocumentHighlightsService .  There's was no need for multiple indirections since we already had one.